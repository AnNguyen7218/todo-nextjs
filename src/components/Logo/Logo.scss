@import 'config.colors';

.logo--dark {
  color: $black;
}

.logo--light {
  color: $white;
}
