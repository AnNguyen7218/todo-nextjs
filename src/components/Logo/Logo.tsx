import React from 'react';
import cx from 'classnames';
import LogoIcon from 'assets/icons/ic-logo.svg';
import css from './Logo.scss';

type Props = {
  width?: number;
  height?: number;
  theme?: 'dark' | 'light';
  className?: string;
};

const Logo: React.FC<Props> = ({
  width = 100,
  height = 27,
  theme = 'dark',
  className,
}) => {
  return (
    <LogoIcon
      width={width}
      height={height}
      className={cx(css[`logo--${theme}`], className)}
    />
  );
};

export default Logo;
