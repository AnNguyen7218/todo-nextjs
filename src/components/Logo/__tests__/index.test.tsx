import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Logo from '../Logo';

afterEach(cleanup);

describe('Logo Component', () => {
  test('Logo should match snapshot', () => {
    const { container } = render(
      <>
        <Logo height={10} width={10} />
        <Logo theme="dark" />
        <Logo theme="light" />
      </>
    );
    expect(container).toMatchSnapshot();
  });
});
