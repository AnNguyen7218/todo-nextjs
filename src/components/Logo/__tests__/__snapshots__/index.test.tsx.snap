// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Component Logo should match snapshot 1`] = `
<div>
  <ic-logo.svg
    classname="logo--dark"
    height="10"
    width="10"
  />
  <ic-logo.svg
    classname="logo--dark"
    height="27"
    width="100"
  />
  <ic-logo.svg
    classname="logo--light"
    height="27"
    width="100"
  />
</div>
`;
